HEC.FDA.Model/structures/Inventory.cs

```diff
@@ -335,20 +339,43 @@ public List<DeterministicOccupancyType> SampleOccupancyTypes(IProvideRandomNumbe
         }
 
 
-        public ConsequenceResult ComputeDamages(float[] wses, int analysisYear, string damageCategory, List<DeterministicOccupancyType> deterministicOccupancyType)
+        public List<ConsequenceResult> ComputeDamages(List<float[]> wses, int analysisYear, string damageCategory, List<DeterministicOccupancyType> deterministicOccupancyType)
         {
-            ConsequenceResult aggregateConsequenceResult = new(damageCategory);
+
+            List<ConsequenceResult> aggregateConsequenceResults = new List<ConsequenceResult>();
             //assume each structure has a corresponding index to the depth
-            for (int i = 0; i < Structures.Count; i++)
+            var structureParallelCollection = new double[wses.Count,Structures.Count];
+            var contentParallelCollection = new double[wses.Count, Structures.Count];
+            var otherParallelCollection = new double[wses.Count, Structures.Count];
+            var vehicleParallelCollection = new double[wses.Count, Structures.Count];
+
+            Parallel.For(0, Structures.Count, i =>
+            {
+                float[] wse = wses.Select(array => array[i]).ToArray();
+
+                for (int j = 0; j < wse.Length; j++)
+                {
+                    if (wse[j] != -9999)
+                    {
+                        ConsequenceResult consequenceResult = Structures[i].ComputeDamage(wse[j], deterministicOccupancyType, PriceIndex, analysisYear);
+                        structureParallelCollection[j,i] = (consequenceResult.StructureDamage);
+                        contentParallelCollection[j, i] = (consequenceResult.ContentDamage);
+                        otherParallelCollection[j, i] = (consequenceResult.OtherDamage);
+                        vehicleParallelCollection[j, i] = (consequenceResult.VehicleDamage);
+                    }
+                }
+                Thread.Sleep(0);
```

Comment:
Is this here for any reason? Sleep for 0 miliseconds?

Reply:
Here's the description of the method parameter in the .net docs "The number of milliseconds for which the thread is suspended. If the value of the millisecondsTimeout argument is zero, the thread relinquishes the remainder of its time slice to any thread of equal priority that is ready to run. If there are no other threads of equal priority that are ready to run, execution of the current thread is not suspended."